tutorials/viewports/using_viewport_as_texture.rst

```diff
@@ -0,0 +1,321 @@
+  }
+
+.. note:: All credit goes to the initial author Inigo Quilez. It is published with the ``MIT`` licence.
+
+Now to use ``noised``, add the following to the  ``fragment`` function:
+
+::
+
+  vec4 n = noised(unit * 5.0);
+  COLOR.xyz = vec3(n.x * 0.5 + 0.5);
+
+.. image:: img/planet_noise.png
+
+.. note:: In order to highlight the texture, I have turned set the material to unshaded.
+
+You can see now that the noise indeed wraps seamlessly around the sphere. Although this
+looks nothing like the planet you were promised. So lets move onto something more colorful.
+
+Coloring the planet
+-------------------
+
+Now to make the planet colors. There are many ways to do this, if you look on `Shadertoy <https://www.shadertoy.com>`_
+you will find all kinds of ways of mapping colors to procedural planet terrain. For now
+we will stick with a simple gradient between water and land. 
+
+To make a gradient in glsl we use the ``mix`` function. ``mix`` takes two values to interpolate
+between and a third parameter to choose how much to interpolate between them, in essence
+it *mixes* the two values together. In other APIs this function is often called ``lerp``.
+Although, ``lerp`` is typically reserved for mixing two floats together, ``mix`` can take any
+values whether it be floats or vector types. 
+
+::
+
+  COLOR.xyz = mix(vec3(0.05, 0.3, 0.5), vec3(0.9, 0.4, 0.1), n.x * 0.5 + 0.5);
+
+The first color is blue for the ocean. The second color is a kind of reddish color (because
+all alien planets need red terrain). And finally they are mixed together by ``n.x * 0.5 + 0.5``.
+``n.x`` smoothly varies between ``-1`` and ``1``. So we map it into the ``0-1`` range that ``mix`` expects.
+Now you can see that the colors change between blue and red. 
+
+.. image:: img/planet_noise_color.png
+
+That is a little more blurry than we want. Planets typically have a relatively clear separation between
+land and sea. In order to do that we will change the last term to ``smoothstep(-0.1, 0.0, n.x)``.
+And thus the whole line becomes:
+
+::
+
+  COLOR.xyz = mix(vec3(0.05, 0.3, 0.5), vec3(0.9, 0.4, 0.1), smoothstep(-0.1, 0.0, n.x));
+
+What ``smoothstep`` does is return ``0`` if the third parameter is below the first and return 1 if the
+third parameter is larger than the second and smoothly blends between ``0`` and ``1`` if the third number
+is between the first and the second. So in this line ``smoothstep`` returns ``0`` whenever ``n.x`` is less than ``-0.1``
+and it returns ``1`` whenever ``n.x`` is above ``0``.
+
+.. image:: img/planet_noise_smooth.png
+
+One more thing to make this a little more planet-y. The land shouldn't be so blobby lets make the edges
+a little rougher. A trick that is often used in shaders to make rough looking terrain with noise is
+to layer levels of noise over one another and various frequencies. We use one layer to make the 
```

Comment:
I think "and" here should be a different word. "levels of noise over one another *using* various frequencies". "at" could work there as well.